admin-dev/themes/new-theme/js/components/seo-preview-card.js

```diff
@@ -0,0 +1,193 @@
+ */
+
+const $ = window.$;
+
+/**
+ * Component is used to preview SEO results.
+ *
+ * Usage: Define HTML for SEO card preview with ID and "data-id-lang" attribute
+ * that is used as default language in which SEO results are displayed.
+ *
+ * <div id="mySeoPreviewCard"
+ *      class="seo-preview-card"
+ *      data-lang-id="1"
+ *      data-seo-preview-url-generator="/admin-dev/generate-seo-preview-url"
+ * >
+ *  <div class="seo-preview-title"></div>
+ *  <div class="seo-preview-url"></div>
+ *  <div class="seo-preview-description"></div>
+ * </div>
+ *
+ * JS: In Javascript you have to enable component.
+ *
+ * new SeoPreviewCard(
+ *    '#mySeoPreviewCard',                            // SEO preview card selector
+ *    'input[name^="category[meta_title]',            // Meta name input selector for SEO preview name
+ *    'input[name^="category[link_rewrite]',          // Friendly url input selector for SEO preview url
+ *    'textarea[name^="category[meta_description]',   // Meta description input selector for SEO preview title,
+ *    'input[name^="category[name]',                  // Name that is used when Meta name is empty
+ *    'textarea[name^="category[description]'         // Description is used when Meta description is empty
+ * );
+ */
+export default class SeoPreviewCard {
+  constructor(
+    seoPreviewCardSelector,
+    titleInputSelector,
+    urlInputSelector,
+    descriptionInputSelector,
+    fallbackTitleInputSelector,
+    fallbackDescriptionInputSelector,
+  ) {
+    console.log(titleInputSelector);
+
+    this.$card = $(seoPreviewCardSelector);
+
+    this.titleInputSelector = titleInputSelector;
+    this.urlInputSelector = urlInputSelector;
+    this.descriptionInputSelector = descriptionInputSelector;
+    this.fallbackTitleInputSelector = fallbackTitleInputSelector;
+    this.fallbackDescriptionInputSelector = fallbackDescriptionInputSelector;
+
+    $(titleInputSelector).on('input', (event) => this._changeHandler(event));
+    $(urlInputSelector).on('input', (event) => this._changeHandler(event));
+    $(descriptionInputSelector).on('input', (event) => this._changeHandler(event));
+
+    $(fallbackTitleInputSelector).on('input', (event) => this._changeHandler(event));
+    $(fallbackDescriptionInputSelector).on('input', (event) => this._changeHandler(event));
+
+    this._refreshCard(this.$card.data('lang-id'));
+
+    return {};
+  }
+
+  /**
+   * Handles even when meta data is changed
+   *
+   * @param {Event} event
+   *
+   * @private
+   */
+  _changeHandler(event) {
+    const langId = $(event.currentTarget).data('lang-id');
+
+    this._refreshCard(langId);
+  }
+
+  /**
+   * Refresh SEO preview card with latest data
+   *
+   * @param {String} langId
+   *
+   * @private
+   */
+  _refreshCard(langId) {
+    if (this.request) {
+      this.request.abort();
+    }
+
+    this.request = $.ajax(this.$card.data('seo-preview-url-generator'), {
```

Comment:
Why not using $.getJSON instead?

Reply:
No reason, I can use `getJSON` if it's any better. :)

Reply:
done.